cmd/kube-apiserver/app/server.go

```diff
@@ -84,8 +84,8 @@ func Run(s *options.APIServer) error {
 	genericapiserver.DefaultAndValidateRunOptions(s.ServerRunOptions)
 	genericConfig := genericapiserver.NewConfig(). // create the new config
 							ApplyOptions(s.ServerRunOptions). // apply the options selected
-							Complete()                        // set default values based on the known values
-
+							SkipComplete()                         // call Complete() after all customizations of the config have been done
```

Comment:
oh,i am sorry. Maybe we could move the MaybeGenerateServingCerts function into Complete(). And call Complete() after all customizations have been done